include/functionref.hpp

```diff
@@ -0,0 +1,238 @@
+// Copyright 2019 The Abseil Authors.
```

Comment:
Minor: Should we put this in third party ?
Ideally we can identify easily what possibly needs to be upgraded / comes from external projects.

Reply:
I think you're right.  We may have other sources that are not properly mentioned in the third party licenses doc.

Reply:
Done, cf. last commit